### DISCUSSION TITLE
Communication between Servers

### DISCUSSION THREAD

**Discussion by francescobianca** [FreeOpcUa/opcua-asyncio#773] [Q&A]
Hi,

I have a set of hierarchical Servers (the lower level one is a subset of the semantic model of the upper one and so on). 
I also have REST APIs that modify the semantic model (add nodes for example). Is there any method with OPCUA to notify the lower servers of the change (For example if I add a node on the main OPCUA Server I would like this to be notified to the lower levels and eventually add the node to the lower levels)?

I read this part of the documentation:
_https://reference.opcfoundation.org/Core/docs/Part1/6.3.7/_

Thanks in advance

**Comment by AndreasHeine** [comment]
The lower servers need to subscribe to the "ModelChangeEvents" (which the upper server has to emit!) and then you need to implement what the lower server should do!

https://reference.opcfoundation.org/v104/Core/docs/Part3/9.32.7/
